be/src/storage/lake/primary_key_compaction_policy.cpp

```diff
@@ -121,13 +133,14 @@ StatusOr<std::vector<int64_t>> PrimaryCompactionPolicy::pick_rowset_indexes(
         rowset_vec.emplace_back(&rowset_pb, stat, i);
     }
     // 2. pick largest score level
-    ASSIGN_OR_RETURN(auto pick_level_ptr, pick_max_level(calc_score, rowset_vec));
+    ASSIGN_OR_RETURN(auto pick_level_ptr, pick_max_level(calc_score && !is_real_time, rowset_vec));
```

Comment:
I think we can remove `calc_score` and always do score calculation within one level?